docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java

```diff
@@ -0,0 +1,90 @@
+package org.testcontainers.containers.startupcheck;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import lombok.SneakyThrows;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.WaitingConsumer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.containers.output.OutputFrame.OutputType.STDOUT;
+import static org.testcontainers.containers.startupcheck.StartupCheckStrategyTest.IndefiniteOneShotStrategyTest;
+import static org.testcontainers.containers.startupcheck.StartupCheckStrategyTest.MinimumDurationStrategyTest;
+import static org.testcontainers.containers.startupcheck.StartupCheckStrategyTest.OneShotStrategyTest;
+
+@RunWith(Suite.class)
+@Suite.SuiteClasses({OneShotStrategyTest.class, IndefiniteOneShotStrategyTest.class, MinimumDurationStrategyTest.class})
+public class StartupCheckStrategyTest {
+
+    private static final String HELLO_TESTCONTAINERS = "Hello Testcontainers!";
+
+    private static void waitForHello(GenericContainer container) throws TimeoutException {
+        WaitingConsumer consumer = new WaitingConsumer();
+        container.followOutput(consumer, STDOUT);
+
+        consumer.waitUntil(frame ->
+            frame.getUtf8String().contains(HELLO_TESTCONTAINERS), 30, TimeUnit.SECONDS);
+    }
+
+    public static class OneShotStrategyTest {
+        @Rule
+        // withOneShotStrategy {
+        public GenericContainer bboxWithOneShot = new GenericContainer("busybox:1.31.1")
+            .withCommand(String.format("echo %s", HELLO_TESTCONTAINERS))
+            .withStartupCheckStrategy(
+                new OneShotStartupCheckStrategy().withTimeout(Duration.ofSeconds(3))
+            );
+        // }
+
+        @SneakyThrows
+        @Test
+        public void testCommandIsExecuted() {
+            waitForHello(bboxWithOneShot);
+
+            assertThat(bboxWithOneShot.isRunning()).isFalse();
+        }
+    }
+
+    public static class IndefiniteOneShotStrategyTest {
+        @Rule
+        // withIndefiniteOneShotStrategy {
+        public GenericContainer bboxWithIndefiniteOneShot = new GenericContainer("busybox:1.31.1")
+            .withCommand("sh", "-c", String.format("sleep 5 && echo \"%s\"", HELLO_TESTCONTAINERS))
+            .withStartupCheckStrategy(
+                new IndefiniteWaitOneShotStartupCheckStrategy().withTimeout(Duration.ofSeconds(3))
```

Comment:
It does seem slightly odd to use a timeout with this strategy - is it intentional?

I'd suggest just:
```suggestion
                new IndefiniteWaitOneShotStartupCheckStrategy()
```
should be appropriate...

Reply:
Will apply and merge.

Reply:
@rnorth I'm sorry for not replying earlier, I was a little busy, but once I saw that you are going to commit I decided to reply straight away :)

That was intentional to show that even though timeout is 3 sec, the command executed on startup won't be interrupted and will continue executing as long as required (which is 5 sec)